code/game/atom/atom_color.dm

```diff
@@ -12,17 +15,19 @@
 	 * its inherent color, the colored paint applied on it, special color effect etc...
 	 */
 	var/list/atom_colours
+	/// Currently used color filter - cached because its applied to all of our overlays because BYOND is horrific
+	var/cached_color_filter
 
 ///Adds an instance of colour_type to the atom's atom_colours list
-/atom/proc/add_atom_colour(coloration, colour_priority)
+/atom/proc/add_atom_colour(coloration, colour_priority, color_type = ATOM_COLOR_TYPE_NORMAL)
 	if(!atom_colours || !atom_colours.len)
 		atom_colours = list()
 		atom_colours.len = COLOUR_PRIORITY_AMOUNT //four priority levels currently.
 	if(!coloration)
 		return
 	if(colour_priority > atom_colours.len)
 		return
-	atom_colours[colour_priority] = coloration
+	atom_colours[colour_priority] = list(coloration, color_type)
```

Comment:
it seems silly to me to need to pass in color_type instead of autodetecting it based off the (is it a listedness + contents) of coloration.